apis/python/src/tiledbsoma/_dask/util.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) TileDB, Inc. and The Chan Zuckerberg Initiative Foundation
+#
+# Licensed under the MIT License.
+
+"""Utilities for loading a TileDB-SOMA ``X`` matrix as a Dask Array."""
+
+from __future__ import annotations
+
+from concurrent.futures import ThreadPoolExecutor
+from functools import cache
+from os import cpu_count
+from typing import (
+    Any,
+    Sequence,
+    TypedDict,
+    Union,
+)
+
+import numpy as np
+import pyarrow as pa
+from numpy import int64
+from numpy.typing import NDArray
+from somacore.options import SparseNDCoord, SparseNDCoords
+from typing_extensions import TypeAlias
+
+from tiledbsoma.options import SOMATileDBContext
+from tiledbsoma.options._soma_tiledb_context import ConfigVal
+
+ChunkSize = Union[int, tuple[Union[int, None], int]]
+JoinIDs: TypeAlias = NDArray[int64]
+
+
+class DaskConfig(TypedDict, total=False):
+    chunk_size: ChunkSize
+    tiledb_concurrency: int | None
+    tiledb_config: dict[str, ConfigVal]
+
+
+def chunk_ids_sizes(
```

Comment:
any reason to not just use numpy.array_split?   (suggestion only in the spirit of writing less code)